ts_src/lib/converter/input/witnessUtxo.ts

```diff
@@ -71,9 +71,11 @@ export function encode(data: WitnessUtxo): KeyValue {
   const nonceLen = nonce[0] === 0 ? 1 : 33;
   const varintScriptLen = varuint.encodingLength(script.length);
   const rangeProofLen = rangeProof ? rangeProof.length : 0;
-  const varintRangeProofLen = varuint.encodingLength(rangeProofLen);
+  const varintRangeProofLen =
+    rangeProofLen === 0 ? 0 : varuint.encodingLength(rangeProofLen);
```

Comment:
```suggestion
    nonceLen === 1 ? 0 : varuint.encodingLength(rangeProofLen);
```

Reply:
The rationale is if `nonceLen` is 1 the witnessUtxo is unconfidential and proofs must not be encoded, otherwise it means it's a confidential one and proofs are encoded whatever their lengths are.